geoproxy/main.go

```diff
@@ -5,19 +5,44 @@ package main
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"log"
+	"math"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
 	"os"
-	"sort"
 	"strconv"
-	"fmt"
+	"strings"
 )
 
-var data = map[int]map[int]string{}
-var sorted_lat []int
+// Coord struct represents the lat-lng coordinate
+type Coord struct {
+	Lat, Lng float64
+}
+
+// Key struct is for map pointing to correct server
+type CoordRange struct {
```

Comment:
objects used as keys have to be comparable. is CoordRange comparable?
https://golang.org/ref/spec#Comparison_operators

Reply:
Yep! Golang seems to be able to compare equality of structs using their field values: https://www.geeksforgeeks.org/structure-equality-in-golang/#targetText=A%20structure%20or%20struct%20in,types%20into%20a%20single%20unit.&targetText=Both%20the%20operator%20and%20method,other%2C%20otherwise%2C%20return%20false.

Reply:
nvm we should be good according the discussion on structs here:
https://blog.golang.org/go-maps-in-action